src/commands/revealResource.ts

```diff
@@ -3,26 +3,56 @@
 
-import { parseAzureResourceId } from '@microsoft/vscode-azext-azureutils';
+import { ParsedAzureResourceId } from '@microsoft/vscode-azext-azureutils';
 import { AzExtTreeItem, IActionContext, parseError } from '@microsoft/vscode-azext-utils';
-import { AppResource } from '@microsoft/vscode-azext-utils/hostapi';
+import { VSCodeRevealOptions } from '@microsoft/vscode-azext-utils/hostapi.v2';
 import { ext } from '../extensionVariables';
 import { ResourceGroupsItem } from '../tree/v2/ResourceGroupsItem';
 import { ResourceTreeDataProviderBase } from '../tree/v2/ResourceTreeDataProviderBase';
 
-export async function revealResource(context: IActionContext, resourceId: string): Promise<void>;
-export async function revealResource(context: IActionContext, resource: AppResource): Promise<void>;
-export async function revealResource(context: IActionContext, arg: AppResource | string): Promise<void> {
-    const resourceId = typeof arg === 'string' ? arg : arg.id;
-
-    context.telemetry.properties.resourceType = parseAzureResourceId(resourceId).provider.replace(/\//g, '|');
+export async function revealResource(context: IActionContext, resourceId: string, options?: VSCodeRevealOptions): Promise<void> {
+    setTelemetryPropertiesForId(context, resourceId);
 
     try {
         const item: ResourceGroupsItem | undefined = await (ext.v2.api.resources.azureResourceTreeDataProvider as ResourceTreeDataProviderBase).findItemById(resourceId);
         if (item) {
-            await ext.appResourceTreeView.reveal(item as unknown as AzExtTreeItem, { expand: false, focus: true, select: true });
+            await ext.appResourceTreeView.reveal(item as unknown as AzExtTreeItem, options ?? { expand: false, focus: true, select: true });
         }
     } catch (error) {
         context.telemetry.properties.revealError = parseError(error).message;
     }
 }
+
+function setTelemetryPropertiesForId(context: IActionContext, resourceId: string): void {
+    const parsedAzureResourceId = parsePartialAzureResourceId(resourceId);
+    const resourceKind = getResourceKindFromId(parsedAzureResourceId);
+    context.telemetry.properties.resourceKind = resourceKind;
+
+    if (resourceKind === 'resource') {
+        context.telemetry.properties.resourceType = parsedAzureResourceId.provider?.replace(/\//g, '|');
+    }
+}
+
+function parsePartialAzureResourceId(id: string): Partial<ParsedAzureResourceId> & Pick<ParsedAzureResourceId, 'rawId'> {
+    const matches = id.match(/^\/subscriptions\/([^\/]*)(\/resourceGroups\/([^\/]*)(\/providers\/([^\/]*\/[^\/]*)\/([^\/]*))?)?$/i);
+    return {
+        rawId: id,
+        subscriptionId: matches?.[1],
+        resourceGroup: matches?.[3],
+        provider: matches?.[5],
+        resourceName: matches?.[6]
+    }
+}
+
+function getResourceKindFromId(parsedId: Partial<ParsedAzureResourceId>): 'subscription' | 'resourceGroup' | 'resource' {
```

Comment:
We should have this return an `AzExtResourceType`.

Reply:
Unfortunately, we have no way of getting the `AzExtResourceType` from the ARM ID, since the `kind` property isn't present.

Should we add subscription, and a generic "resource" type to `AzExtResourceType`?

Reply:
Subscription makes sense to have a type for, IMO. I don't think a generic resource though makes sense. Though, since all of this is internal/private, I don't think it matters too much.